src/services/playlistManagementService.js

```diff
@@ -50,12 +55,20 @@ function getManagedPlaylistsIds(refreshToken) {
 }
 
 async function managePlaylist(playlistId, refreshToken) {
-  await validatePlaylistBelongsToUser(playlistId, refreshToken);
+  await module.exports.validatePlaylistBelongsToUser(playlistId, refreshToken);
+  const lockKey = `${refreshToken}-${playlistId}`
```

Comment:
I'm not sure why use refreshToken as the lock key. If the token changes while a reordering is in progress, isn't there a chance of overlap?

Reply:
You are totally right, this was a mishap the lock key should be something unique such as the playlistId